### ISSUE TITLE
[Problem]: After switching to CC2652 coordinator, my network works perfectly, but no device will pair with it.

### ISSUE THREAD

**Issue by convicte** [Koenkk/zigbee2mqtt#10339]
### What happened?

I transitioned from an old and weak CC2531, which worked ok, but with 40 devices connected started to misbehave, lose connection to devices, etc. Unfortunately, upon switching  to CC2652 there is no way to pair any new device (IKEA, XIAOMI, TUYA, etc.). The network rebuilt perfectly fine after the coordinator change, and is updating sensors just fine, but will not execute a pairing process.

I've changed back and forth between different coordinator firmwares, to no effect.
Unplugging the coordinator, restarting the VM in which HA is running, neither restarting HA made any effect.

### What did you expect to happen?

When pairing is initiated, the interview process begins and the device joins the network as per the old CC2531.

### How to reproduce it (minimal and precise)

1) Connect the next adapter to the network
2) Permit_join enabled
3) Put device in pairing mode
4) Wait until the device times out because nothing will happen

### Zigbee2MQTT version

1.22.1

### Adapter firmware version

20211217, 20210319, 20210708

### Adapter

Ebyte CC2652P

### Debug log

How to record a debug of something that doesn't happen? I've been monitoring the logs and while the permit_join is enabled, no interview process is initiated. The device sits idle until it times-out.

**Comment by Koenkk** [comment]
try with the fw from https://github.com/Koenkk/Z-Stack-firmware/tree/develop/coordinator/Z-Stack_3.x.0/bin

**Comment by convicte** [comment]
I am on it now - no luck. The exact same behavior as on the two others. 

The network works great, but pairing is completely dead.
![image](https://user-images.githubusercontent.com/78666770/147295340-17b7a5fc-90ff-4465-b4d5-b20557b49970.png)

**Comment by henkiejan1** [comment]
Hmm, i have recently switched to the CC2652 also but no problems at all with repairing some devices. Did you delete devices who exist in your network and try to add again? If its not a new device maybe exist the device just in your configuration.yaml?

**Comment by Tonio16** [comment]
Hello

I switched from a zigate to a sonofff itead yesterday. No pairing issue with this firmware : 20210120


Antoine

**Comment by Koenkk** [comment]
@convicte 
- Could you make a sniff when trying pairing this device? https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#with-cc2531
- Could you share your `coordinator_backup.json` here?

**Comment by convicte** [comment]
@Koenkk 

Thanks for the update and my apologies for the late reply.

1) If I understand correctly from the link attached, you are asking to use the old CC2531 I have, with a sniff firmware plus a windows tool to collect a sniff of the network traffic at my selected channel - is that correct?
2) Please see the backup below (minus the keys - obviously).
![image](https://user-images.githubusercontent.com/78666770/147395723-7b45b153-5073-4b8b-b4c9-ba4dc73dc3bd.png)


One more data point for the evaluation; I've been trying to pair another IKEA Dimmer and held the reset button for about 10 seconds, while in the room with a Zigbee floor lamp. At the same time, this lamp was kicked out of my network with the following log output and can't be repaired, for obvious reasons - because nothing else can be.
![chrome_6QA8yAcywg](https://user-images.githubusercontent.com/78666770/147395700-168e1833-98a6-4a55-8852-3e9308965326.png)

**Comment by Koenkk** [comment]
> If I understand correctly from the link attached, you are asking to use the old CC2531 I have, with a sniff firmware plus a windows tool to collect a sniff of the network traffic at my selected channel - is that correct?

Yes

>  I've been trying to pair another IKEA Dimmer and held the reset button for about 10 seconds, while in the room with a Zigbee floor lamp (Office floor lamp).

See https://www.zigbee2mqtt.io/devices/ICTC-G-1.html#pairing how to reset this device, when you hold it 10 seconds it will setup a new network with just the dimmer and the lamp

**Comment by convicte** [comment]
> > If I understand correctly from the link attached, you are asking to use the old CC2531 I have, with a sniff firmware plus a windows tool to collect a sniff of the network traffic at my selected channel - is that correct?
> 
> Yes

My sincere apologies, but I am failing at providing a sniff. I've spent quite some time to get the CC2531 set up as the sniffer tool, but having no CC debugger and unable to find a way to flash it via USB as I did for the 2653 using the TI Flasher V2 software, I am a bit stuck. 
I've downloaded the ZBOSS V.2 since V1.0 doesn't provide the sniffer.hex firmware file, but can't get further than this.
Some guidance would be greatly appreciated!

Is there any chance my backup file gave some hints?

For your kind consideration, my z2m config as well:
![image](https://user-images.githubusercontent.com/78666770/147468826-d9b489d3-2dae-4283-972d-cb456db520ee.png)
Please note, I change the permit join from the HA interface, but I wonder if I should set it permanently to true in the config and switch off from the interface (or vice versa)

> 
> > I've been trying to pair another IKEA Dimmer and held the reset button for about 10 seconds, while in the room with a Zigbee floor lamp (Office floor lamp).
> 
> See https://www.zigbee2mqtt.io/devices/ICTC-G-1.html#pairing how to reset this device, when you hold it 10 seconds it will setup a new network with just the dimmer and the lamp

There may have been a misunderstanding in that I was not trying to pair it this way. I know it's to be reset by pressing the button 4 times - https://www.zigbee2mqtt.io/devices/E1743.html (it's actually this IKEA dimmer/switch).
In my desperation, I was trying different things to make the network recognize it. 

All this said, I am still unable to explain why getting the IKEA button into it's direct pair mode would kick out a different brand device from the network, as seen above. 

When I fix pairing issues I'll play with this, but for the moment getting something to pair with my network is a priority.

**Comment by Koenkk** [comment]
@convicte unfortunately I need a sniff to debug this further, one thing you can try is to pair the dimmer when holding it very close to the coordinator (to ensure it tries to pair via that)

**Comment by Fabiancrg** [comment]
@Koenkk I had some issue myself recently to pair devices.
The only way to do it was to pair them close to the coordinator (<1m), when I was too far it was not working anymore.
I experienced it just now with two devices, one Aqara contact sensor and one Tuya roller shutter  I was not able to pair.

I tried the new FW on my CC2652R and the pairing worked just fine form their place (not close to the coordinator).

**Comment by convicte** [comment]
@Koenkk 
My apologies for the late reply. It was a busy week.

1) In desperation, having no way to flash the sniffer I borrowed a Sonoff 3.0 dongle which was running an interely different firmware from July 2021, and it exhibited the same exact behavior - perfect network performance but inability to pair devices, so it's unlikely to be my hardware
2) I tested 5 different devices, both holding it next to the coordinator and in an entirely different part of the house, via routers. Did not make any difference. I once managed to get the device associated with the network, but it failed to recognize and when removed never associated again, like the others.
3) I've also changed the USB extension, which is used to get the dongle outside my server rack, for better reception. Two different extensions didn't make a difference.

Finally, I started looking for ways how to flash the sniffer with available hardware I had and found an old PI and instructions how to use it to flash a CC2532 which worked!!

I now have a sniffer and managed to follow your instructions to grab a paring process via routers with an IKEA button listed above. My only concern with posting it, hence it will contain network keys. Is there any way to share it with you securely?
Second, I can grab another pairing process right next to the coordinator, if this is of interest?

Thank you!!

**Comment by Koenkk** [comment]
> Second, I can grab another pairing process right next to the coordinator, if this is of interest?

yes

You can send the network key to me on telegram (@koenkk)

**Comment by convicte** [comment]
Thank you!

I just texted you on Telegram with the key.

[Pairing attempt via router.zip](https://github.com/Koenkk/zigbee2mqtt/files/7797915/Pairing.attempt.via.router.zip)

I'll upload the second file with the coordinator direct capture later today.

**Comment by convicte** [comment]
Thank you!

I just texted you on Telegram with the key.

[Pairing attempt via router.zip](https://github.com/Koenkk/zigbee2mqtt/files/7797917/Pairing.attempt.via.router.zip)

I'll upload the second file with the coordinator direct capture later today.

**Comment by Koenkk** [comment]
I've checked the coordinator sniff, somehow the joining device isn't doing an association request. But while checking your sniff I found another issue. It seems the coordinator is using a different epid than the rest of the network leading to pan conflicts.

<img width="1021" alt="Screenshot 2022-01-02 at 11 21 16" src="https://user-images.githubusercontent.com/2892853/147872886-1fcf9f62-e372-4034-85c7-b423540146a3.png">


@castorw my first thought is that this is caused by the cc2531 returning the wrong epid (0xdddd..), during the migration we write the wrong epid into the cc2652, could that be the case here?

**Comment by convicte** [comment]
Thank you! I've seen this conflict in the sniff but failed to recognise the significance.
I've never touched the panID during the migration but I am wondering how it could get changed if it's listed in the z2m configuration, which should have been burned in to the CC2652 upon first boot. Can I change it now on the CC2652 or attempt to correct it when the backup is loaded onto a fresh CC2652 (after flash wipe)?

Just in case you are wondering I could run another sniff with a different device entirely, to see if it's also not sending out a pairing request.
This said, I am at a loss why changing the coordinator would case the endpoints not to request pairing.

My apologies if these are trivial questions.

**Comment by Koenkk** [comment]
@convicte can you check if the `extended_pan_id` in the `coordinator_backup.json` is also `dddddddd...`? If yes, change it to the one starting with 95e7 in the sniff, reflash your coordinator and start z2m might solve this problem.

**Comment by convicte** [comment]
Yes, indeed it is!
![image](https://user-images.githubusercontent.com/78666770/147888555-b0e73d11-3eb1-4c06-82ec-0f23b59a103e.png)

I'll change it in a moment. By reflash you mean the latest 20211217 firmware, to clear the 'old' backup on the coordinator and make it pull the new one with the correct `extended_pan_id` from the `.json` file?

Is that correct?

**Comment by Koenkk** [comment]
- stop z2m
- update extended_pan_id in the coordinator_backup.json
- reflash with `20211217`
- start z2m

**Comment by convicte** [comment]
Perfect, I did half of that already, so just need to run into the attic to get the stick down, to reflash it.

I'll update everyone ASAP.

**Comment by convicte** [comment]
Quick update:

1) Following the changes, there is definitely progress, but it's not completely resolved yet. I am predicting you will request another sniff of the situation as it stands.
2) When standing right next to the coordinator, the same IKEA remote successfully announced itself, and started the joining process, but eventually left the network. This happened twice, with the third try resulting in the remote remaining in the network but as an unsupported device. Below see the excerpt from the z2m logs of one such event:

`debug 2022-01-02 21:52:09: Received MQTT message on 'zigbee2mqtt/bridge/request/permit_join' with data '{"device":null,"time":254,"transaction":"7fshf-1","value":true}'
info  2022-01-02 21:52:09: Zigbee: allowing new devices to join.
info  2022-01-02 21:52:09: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"7fshf-1"}'
info  2022-01-02 21:52:25: Device '0xccccccfffe020fd9' joined
info  2022-01-02 21:52:25: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xccccccfffe020fd9","ieee_address":"0xccccccfffe020fd9"},"type":"device_joined"}'
info  2022-01-02 21:52:25: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"0xccccccfffe020fd9"},"type":"device_connected"}'
info  2022-01-02 21:52:25: MQTT publish: topic 'zigbee2mqtt/0xccccccfffe020fd9/availability', payload 'online'
info  2022-01-02 21:52:25: MQTT publish: topic 'zigbee2mqtt/0xccccccfffe020fd9', payload '{"last_seen":"2022-01-02T21:52:25+01:00"}'
info  2022-01-02 21:52:25: Starting interview of '0xccccccfffe020fd9'
info  2022-01-02 21:52:25: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xccccccfffe020fd9","ieee_address":"0xccccccfffe020fd9","status":"started"},"type":"device_interview"}'
info  2022-01-02 21:52:25: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"0xccccccfffe020fd9"},"type":"pairing"}'
info  2022-01-02 21:52:27: MQTT publish: topic 'zigbee2mqtt/0xccccccfffe020fd9', payload '{"last_seen":"2022-01-02T21:52:27+01:00"}'
info  2022-01-02 21:52:27: MQTT publish: topic 'zigbee2mqtt/0xccccccfffe020fd9', payload '{"last_seen":"2022-01-02T21:52:27+01:00"}'
debug 2022-01-02 21:52:27: Device '0xccccccfffe020fd9' announced itself
info  2022-01-02 21:52:27: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xccccccfffe020fd9","ieee_address":"0xccccccfffe020fd9"},"type":"device_announce"}'
info  2022-01-02 21:52:27: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0xccccccfffe020fd9"},"type":"device_announced"}'
warn  2022-01-02 21:52:36: Device '0xccccccfffe020fd9' left the network`

3) Trying to achieve the same away from the coordinator amounts to nothing - no messages or response from the interface when the device is in pairing mode. Unable to pair any other device, either, when away from the coordinator.
4) The whole network restarted and once again is working perfectly as is.

Let me guess, a sniff of pairing next to the coordinator and one via routers? ;)

Cheers!!

**Comment by Koenkk** [comment]
Can you provide a sniff when pairing a non battery powered device (something like a switch or light) via a router?

**Comment by convicte** [comment]
Absolutely!

Please see pairing attempts of the same lamp as listed above (Lidl Livarno).
It seems to be stuck in a pairing loop where association is successful, but it's actually not showing up in the interface at all.
No messages or iconography is depicted in the z2m interface at all. It's as if the process never happened, as far as the HA interface is concerned.

[Pairing attempt via router for a floor lamp.zip](https://github.com/Koenkk/zigbee2mqtt/files/7799818/Pairing.attempt.via.router.for.a.floor.lamp.zip)

Same passwords as yesterday.

**Comment by fuglphoenix** [comment]
@convicte, This sounds a lot like my problem. 

I have a Slaesh's CC2652RB Coordinator and just updated to 20211217. 
I haven't migrated, but did a fresh setup when i recieved the stick.
z2m version is 1.22.2

i'm trying to pair my devices (ikea, Tuya and others) via a ikea router, since they are a bit far away from the coordinator.
it seems to work since the devices are stopping the light blinking sequence (join mode) but when i look in z2m no device are showing up. 

if i disable join in z2m and turn the device off and on. It will revert back to the light blinking join mode.
if i move them, i can pair them directly to the coordinator

unfortunately i haven't got any spare zigbee stick to sniff what is happening.

**Comment by vadimag1** [comment]
I have exactly the same problem. Devices are not connected to the coordinator.

**Comment by cracrama** [comment]
Hmm similiar situation with my network - after flashing 20211217 on Ebyte CC2652P I have strange behaviour from couple of sonoff devices - temp humidity sensors. I can pair them but after pairing they not reporting anything and finally disconectiong from network. However one of these devices - same model still works and reporting correctly - there was no need to repair. Other devices pair / report correctly. I want to also mention that it seems like affected are devices which were disconected from network after fleshing firmware. I don't know what happened and that behaviour is really strange.

**Comment by Coax88** [comment]
Hi
Im experience the same issue after migrate from cc2531 to cc2652p, before I could pair without any problems, after upgrading i have issue parring and need alot of times take device close to the sonoff cc2652p before it pairs.

**Comment by xrust83** [comment]
The problem is more global and does not seem to be related to the adapter firmware. I'm still using ss2531 I flashed on the latest firmware Z-Stack_Home_1.2 20211115/20211116 got the same result it seems to work but I can not connect a new or remote old device. If the problem was only in the firmware, then rollback and the old version of Z-Stack_Home_1.2 20201127/20201128 would return everything back, but it does not happen, so the reason is in the version itself zigbee2mqtt

**Comment by cracrama** [comment]
I have reflashed firmware with developed branch and it seems like everything it's fine - I had to repair sonof end devices and then everything seems fine. Word of advice - Once you reflash firmware let the network to "settle down" couple of hours because it seems like that plays a big role.

**Comment by Coax88** [comment]
I it helped for me to change zigbee channel away from other wifi channels, now it work perfect and can pair without any problems

**Comment by jnxxx** [comment]
I have somewhat similar issues as others seem to.
I first used a CC2531 stick, which actually had worked fine. But in December I started to experience issues as more devices were added (~15).
Eventually a Sonoff temp and humidity device stopped updating. Although I could re-pair it, it would only send one update then be stuck again on that value.

I figured it was the limitation of the CC2531, and I ordered a CC2652RB which I got yesterday and flashed with Z-Stack_Home_3.x.0_20211217.
The Sonoff device is now paired and seem to work again.

However, I now experience problems adding new devices (Aqara door & window contact sensors).
I usually do so by permitting join to a specific router. But this does not work and does not seem to log anything.
If close enough I _can_ pair it to the coordinator, but that is it. As soon as I try to pair it to a router, it leaves the network and does not get connected to the router.

Otherwise things seem to be communicating as is.

**Comment by cracrama** [comment]
What i have done so far: Flashed CC2652P Ebyte with new firmware and whereas most devices reconnecting correctly, some (like Sonoff, aquara) none. OK I have reflashed again and same outcome - I have reseted Z2M but nothing. What is interesting it's that it's same device causing problem every time no matter firmware - pairing ok but no reporting or only partial reporting. So I will agree with xrust83 that it cound be global problem. I have tried different dongle - same story - even same type but different device and same story., I need to mention that devices befor flashing firmware were working ok and some of them are brand new.

**Comment by xrust83** [comment]
I couldn't beat the cc2531 but was ready before switching to the cc2652p.  I started flashing to connect Aqara MCCGQ14LM.  after switching to cc2652p, the devices are paired correctly, but the Aqara MCCGQ11LM sensor began to fail.  Somewhere in the topics I met the answer that the problem is in the CH340C port driver, something is wrong in updating the OS drivers.  Therefore, I temporarily transferred this sensor to the Aqara hub.  It is also worth adding that lowering the ss2531 firmware version until the past few days has not repaired the problem with connecting new devices or old remote ones.  so I still think that the problem is not in the firmware of the stick, but in the program itself.

**Comment by Koenkk** [comment]
For users that migrated from a CC2531 to a CC2652 we found the possible issues; keep an eye on https://github.com/Koenkk/zigbee2mqtt/issues/9117#issuecomment-1015409585

**Comment by cracrama** [comment]
Many thanks for direction
its just firmware update of the same chip caused this kind of issues (CC2652) too Seems like even downgrading firmware doesn't fix the issue  I can also confirm that switching gateways (Texas instrument / Ebyte ) doesnt change anything. 
Many thanks for support Gents. I will try more experiments and we will see.

**Comment by sonsaeng** [comment]
I'm sorry, same Problem. I use a Raspi stand alone with sonoff Z3 Dongle. Only updated raspian-lite, MQTT, Z2M and Nodered.
Every seems to be ok, but no pairing. Sonoff via extender and powerd usb-hub. No zigbeedevice (it would be the first one) will connect.

**Comment by Tonio16** [comment]
@sonsaeng
Which firmware do you have on the controler ? 
It has to be the one of December 21 or January 21.

Antoine

**Comment by sonsaeng** [comment]
I used this:  CC1352P2_CC2652P_launchpad_20210120

**Comment by castorw** [comment]
I believe this is related to #9117. I will be investigating further into this issue as per https://github.com/Koenkk/zigbee2mqtt/issues/9117#issuecomment-1016890665.

**Comment by sonsaeng** [comment]
I solved my problem in this way:
- with the TI Flash_Programmer I flashed the Sonoff 3.0 USB Dongle with:    CC1352P2_CC2652P_launchpad_coordinator_20211217 
- I changed the channel in the /opt/zigbee2mqtt/data/configuration.yaml from 11 to 25
- I deleted: coordinator_backup.json & database.db & state.json
- I stopped Z2M
- npm start

Now I could pair an IKEA-lamb (router) and a tradfri dimmer (enddevice) :)

Thank you for your help!

**Comment by cracrama** [comment]
sonsaeng So is this operation required repairing all devices again?

**Comment by sonsaeng** [comment]
Yes, thats what I have understood. You have to pair them again. I'm no expert in zigbee (I only use it). But what I have seen is, that the data-structure has changed a lot. I think thats the reason why. I know that this means a lot of work. But if things work better than, it`s ok for me.

**Comment by convicte** [comment]
Dear all,

I've been delinquent on updating this ticket with the outcome of my troubleshooting and finding a final solution to the problem, which is entirely thanks to @Koenkk and his valuable input - Thank you again Sir!

The issue, at least as it manifested itself for my network and what were the root causes of it did not require changing channels, deleting your entire network nor did it require repairing of most devices, but a few most stubborn had to be reset and repaired.

Main issues in question - the devices will not pair anywhere in the network, but will right next to the coordinator:
1) The coordinator_backup.json in your existing network for some reason does not include all the routers (e.g. they were not added over time as you added them to the network, and thus the backup did not bring them into your new coordinator as viable connection points. This results in the pairing failing as the endpoint devices try to use these routers to get to the coordinator. 
2) Repairing the routers closest to the coordinator and then following down the network to the furthest allowed me to get sufficient number to heal the whole network and allow pairing from anywhere to anywhere. 
3) To accomplish this, you will have to repair routers which form the core of your network and delete the backup file from HA z2m folder prior to removing the coordinator and reflashing it with your firmware of choice. I am running the latest v20211217.
Upon repluging the coordinator and lunching z2m the new backup (including your routers) will be loaded onto the coordinator and this issue should be resolved.

Additional issue present in my instance - extended_pan_id in the coordinator_backup.json incorrect or blank:
See here for the troubleshooting, and resolution: https://github.com/Koenkk/zigbee2mqtt/issues/10339#issuecomment-1003693981

In short, if you find your ext_pan_id to be incorrect or blank as was my case, the steps to resolve it overlap with the resolution of the main issue above, though if the issue is combined, like it was for me, you will need to reflash twice.

In short: 
- stop z2m
- update extended_pan_id in the coordinator_backup.json
- reflash with 20211217
- reconnect it
- start z2m
- repair some routers next to the coordinator - you need it to make them recognize endpoint devices when pairing from afar
- delete the backup.json file
- let it recreate, which happens several times a day or upon boot of z2m
- stop z2m
- reflash with 20211217
- replug the coordinator
- start z2m
- let it boot and get the backup loaded to the coordinator
- leave the network to run for 2-12 hours (depending on the size) to heal itself
- repair whatever doesn't get updated/repaired automagically
- the end

If you have any questions, I'll do my best to help, but this is mostly what I've learned while working with Koen, so anything more involved may be way above my understanding.

Good luck!!

**Comment by Fabiancrg** [comment]
On my side, the problem was fixed by just flashing the coordinator with FW 20211217, suddenly a router (sitting far from the coordinator) I was trying to join for months joined a few seconds later after allowing new devices to join, without doing anything else.

But if I look my network, I have a total of 69 devices in which 27 are routers.
In the coordinator_backup.json file I have only 35 devices in which 3 are not present on my network and only 12 routers.

@Koenkk is it normal ? Should I try to remove and pair these missing routers ?
What should be present in the coordinator_backup.json file ?

To be clear, everything seems to work fine, devices are reporting regularly and working fine, I can pair devices too.

**Comment by cracrama** [comment]
convicte - Many thanks for instruction 
However it seems like pairing is not the only  issue. Also reporting - devices pairing with no problems but they are not transmitting any data or transmiting only some. Some devices which are not reporting after couple of hours disconnecting from the network. I have not decided yet, but i thinking about reflashing coordinator and deleting backup. I just hope that will fix the problem and not make it worst. Anywa many thanks

**Comment by convicte** [comment]
@Fabiancrg, from what I understand, ALL the routers should be present in the backup file.
If they are not, I believe that the pairing process through this router to the coordinator will not work.
I may be wrong, though.

@cracrama, it may be a different manifestation of the same problem. If the device is not able to reach the coordinator and is relying on the router relaying the signal to the coordinator, while this router is not recognized as part of the network (as discussed above) the reporting may be sketchy or impossible. Please confirm how your ext_pan_id looks like and what your backup has listed. If the issues match up with mine, you will probably have to follow the above.
Finally, I've had devices do this as well, where they would drop out after some time without reporting, which was related to the above issue.

Based on Koen, this must thankfully only be done once as you transition from Zigbee 1.2 to 3.x, and will not be an issue as the 3.x network starts working correctly. The 3.0 seem to have many more checks and security features which do not allow the looser, potently error laden 1.2 spec networks to transition seamlessly.

**Comment by cracrama** [comment]
convicte. I have checked  it's not pan id problem (mine is "extended_pan_id": "dddddddddddddddd").
And as i have mentioned above i just updated dongle firmware  (no migration) but now even when i reflash dongle or use different dongle i'm experiencing same issues. So probably issue is with Z2M backup_json or something. One observation though: problematic device has " tx_counter": 0" So it's seems like it's not talking. I can force remove device and later re-pair but once device became problematic - nothing changes. Relateing to routers - I have correct reporting from all of them so I'm not sure if that is the same issue.

**Comment by castorw** [comment]
@cracrama Zeroed `tx_counter` for APS link keys is normal and expected with current Z2M implementation. If `rx_counter` is non-null it means the device is transmitting APS-encrypted messages but the coordinator is not responding with encrypted messages. Do any other of your devices have `tx_counter` larger than zero? I can investigate further if you want but I will need you to have a ZB sniffer ready and you can contact me on Telegram (@castorko).

**Comment by cracrama** [comment]
castorw  - Many thanks for response 
rx_counter is null on all devices, however tx_counter has higher values on all devices except that "problematic" where both positions are null. 
Regarding ZB sniffer - I will try to organize something if I find time (I'm quite fresh in zigbee). 
Many thanks anyway.

**Comment by convicte** [comment]
@cracrama, you seem to be misunderstanding the pan_id issue.

From what you described, you have exactly the same problem as I had. The id of ddddddddddd... is what I referred to as blank, since it should look like the one here in the sniff (EPID).

If the coordinator and the network have different values, you will have exactly the issues you described.

https://github.com/Koenkk/zigbee2mqtt/issues/10339#issuecomment-1003693981

**Comment by Fabiancrg** [comment]
@convicte So I deleted and paired all routers not present in the backup file.
After that I restarted Z2M to see the content of this file and found that 10 end devices are still missing.
After a few hours I can tell that these 10 devices are not reporting anything anymore so I will pair them back.

Concerning the EPID, I also have dddd... so I guess I will have to sniff my network to see if I have PAN conflict.

**Comment by Fabiancrg** [comment]
@convicte Is it really a problem having an EPID equal to dddddddddddddddd ?

Mine is like that but I have just ran a trace and and I can see this:

![EPID](https://user-images.githubusercontent.com/22480391/150489869-62af4001-0d79-4024-98ab-bf858b04065c.png)

It looks OK to me as it's the same EPID on the coordinator and on the other devices on my network so I don't see any PAN id conflict either.

**Comment by LCerebo** [comment]
I've done the upgrade from a CC2531 to a sonoff 2652P, everything worked fine except for the pairing process of a device. I've sniffed the traffic and I notice that the coordinator is using a new EPID, but the already paired devices used the "empty" EPID: dddddddddddddddd. 
Turning off z2m, re-flash the coordinator, change the "extended_pan_id" to dddddddddddddddd, and start z2m solved the problem, for the moment. Now I can repair the devices, but some of them are using the other pan_id and I'm not able to pair them. 
Any idea on how to solve this conflict?

**Comment by convicte** [comment]
@Fabiancrg, if you don't have a conflict, I would assume everything is fine, as long as all your devices are on the same EPID.
Outside that I must defer to @Koenkk since it's beyond my expertise.

@LCerebo I am glad the troubleshooting steps worked for you!
In my case, all endpoint or router issues were resolved after forcibly removing them from the network and repairing.
If these are not able to repair, are you also having the other problem where the router you try to pair to is not recognized in the backup file?

**Comment by Fabiancrg** [comment]
@convicte All my issues were fixed when I applied your procedure and removed/paired all routers that were not in the coordinator_backup.json file.

So I would like to help others and create a script that will scan three files:
1. configuration.yaml
2. database.db
3. coordinator_backup.json

The output will list all missing routers with the friendly name, those ones will need to be removed/paired again
Missing end devices with friendly names, same actions to perform but as a second steps as some might back after pairing the missing routers.
Extra devices... I currently don't know what to do with these, I can't force-remove them.

@Koenkk is there a way to force the generation of the coordinator_backup.json without restarting Z2M ? I think it will be better if the script can run without causing any downtime of Z2M.

Any idea if the conflict EPID could be detected without snifing the Zigbee network ?

**Comment by LCerebo** [comment]
> @LCerebo I am glad the troubleshooting steps worked for you! In my case, all endpoint or router issues were resolved after forcibly removing them from the network and repairing. If these are not able to repair, are you also having the other problem where the router you try to pair to is not recognized in the backup file?

Yes, all the routers that use the empty pan_id "dddddddddddddddd" aren't listed in the coordinator_backup.json. 
I tried to repair them but they are listed as "unsupported devices". I don't know what to do to fix the pan_id conflict.

**Comment by convicte** [comment]
Sorry to sound like a broken record, but you have to follow the steps I outlined above.
If the routers are not listed, they are not considered in the pairing process, and thus your network will be handicapped until this is resolved.

Review the routers needing to be repaired, starting from the one closest to the coordinator, and take it from there. In the meantime, please reflash the coordinator as indicated above.

If you have any specific questions or concerns, please let us know. Hopefully someone will be able to help if I fail to do so.

**Comment by fuglphoenix** [comment]
I can confirm that i had the same problem with the EPID. 
in my configuration it wasn't set to anything and it then defaulted to dddddd.... which I confirmed in Settings->RAW
but for some reason it was changed in my coordinator_backup.json to the same value as my coordinator_ieee ?!
I proceeded to stop Z2M, changed the EPID in my coordinator_backup.json reflashed with CC2652RB_coordinator_20211217 and started Z2M again.
now it works and I can pair my devices through a router which was my initial problem

**Comment by MartB** [comment]
Can someone please tell me in simple terms what my EPID is supposed to be and how i get it to stick?

Config:
```
"ext_pan_id": [
                221,
                221,
                221,
                221,
                221,
                221,
                221,
                221
            ],
```

Coordinator backup:
```
"extended_pan_id": "00124RESTOFIEEE",
```

I did what @fuglphoenix did, but reflashing does nothing, extended_pan_id keeps being my coordinator ieee even if i change the backup and re-flash.

**Comment by fluffymadness** [comment]
my coordinator_ieee and extended_pan_id  are completely the same, is this as it should be or should these two be different?

**Comment by arjanvanleent** [comment]
Hello my "extended_pan_id": "dddddddddd", and not all end device are in the backup.
How do I find the correct "extended_pan_id" ?
Don't have another stick to snif
Is it possible to use another installation?
- clean the stick by reflashing it
- Put it in another installation pair 1 end device
- Check the new backup for the correct  "extended_pan_id":
Can this work also?

**Comment by drogfild** [comment]
My story here. I could not pair new devices reliably, especially Ikea E1743 were failing. Some router devices behaved strange during joining; they removed them selves from the network after interview. 

My coordinator is Smartlight CC2652P, don't have original fw version info anymore but it was stock from manufacturer.

I did EPID change to dddddddddddddddd with this guide https://github.com/Koenkk/zigbee2mqtt/issues/10339#issuecomment-1017521526 
Also did those two reflash with 20211217
That dddd... is my guess as I don't have siffing device. But previously I was CC2531 so I took a wild guess.

That didn't seem to resolve any of my pairing issues. I then flashed dev version CC1352P2_CC2652P_other_coordinator_20220103

At least for now that seems to resolve my issues. I was able to pair those Ikea E1743 and all other devices seem to behave now correctly. This as an information to all others.

**Comment by convicte** [comment]
@drogfild 
Not sure if I understand you correctly, but changing from something TO dddd... is the exact opposite of my instructions.
If I was not clear, my apologies. 
Considering it seems to work, you may have done something that helped, but it doesn't seem to be similar to my situation.
If I were you, I would look into the backup file, to see if the routers are there, but that's just me.
@arjanvanleent 
I am sorry, but I cannot help in this regard.
The sniff to look for the EPID was the only route I was informed of.

**Comment by drogfild** [comment]
Thanks @convicte . In my case I had router in the backup file before all this and also afterwards.
It might be that all jumps and hoops with EPID was not needed for me. And new FW alone would have done the needed corrections for me. But as I don't have a sniffer I'm not able to tell. It was wild guess and maybe it was correct or maybe it wasn't, but I'm happy now that everything seems to be working.

There's pretty similar case about EPID with dddd.. at https://github.com/Koenkk/zigbee2mqtt/issues/9117#issuecomment-1017479712

It would be really cool to have some other way to see EPID conflicts than sniffing them. But that's most probably a limitation of zigbee network structure.

**Comment by Justahobby01** [comment]
TLDR: z2m is a dumpster fire. Move to zha.

**Comment by mihaimdinca** [comment]
I have the same issue with the CC2652P. Everything pairs with it but not with any other router in my ZigBee network. I found the discussion here pretty confusing and I was hoping that someone who actually managed to fix the problem could wrap things up in a sort of step by step tutorial for us noobs. Thank you in advance.

**Comment by Fabiancrg** [comment]
@mihaimdinca, if you have the same problem as I had, t's due to some router devices that are define correctly in Z2M.
They might be working properly, e.g.: a bulb might be working but not forwarding the messages to the coordinator.
You can try to remove and pair every router device one by one or look for the one(s) not defined correctly.
For this, I did the python script bellow, you can execute it and it will list the routers that you will need to remove and pair again.

Copy the following code in Z2McheckDevices.py and execute it using python3 Z2McheckDevices.py
````
import` json
import yaml


devices = {}     # Dictionnary with all devices
missingRouters = {}
missingEndDevices = {}
additionalDevices = {}

backupfile = open('/opt/zigbee2mqtt/data/coordinator_backup.json',"r")

DBfile = open('/opt/zigbee2mqtt/data/database.db', 'r')
Lines = DBfile.readlines()


# Parse database.db to retrive all devices IEEE address and type define in Z2M

print('Parsing database.db file')

for line in Lines:
    DBdata = json.loads(line)
    if "ieeeAddr" in DBdata and "type" in DBdata:
        if DBdata ["type"] != 'Coordinator':
            devices[DBdata["ieeeAddr"]] = [DBdata ["type"], "", False]
            #print(DBdata["ieeeAddr"] + " is a " + DBdata ["type"])

DBfile.close()
print('Parsing database.db file... Done')

# Parse coordinator_backup.json to retreive all IEEE present in coordinator

data = json.loads(backupfile.read())


for device in data['devices']:
     dev = devices.get("0x" + device['ieee_address'])
     if dev:
         devices["0x" + device['ieee_address']][2] = True
     else:
         devices["0x" + device['ieee_address']]=["Unknown","", True]

backupfile.close()
print('Parsing backup file... Done')

# Retreive friendly name from configuration.yaml

configfile = open('/opt/zigbee2mqtt/data/configuration.yaml',"r")

configData = yaml.safe_load(configfile)
configDevices = configData["devices"]


for key,value in devices.items():
    dev = configDevices.get(key)
    if dev:
        devices[key][1]=configDevices[key]["friendly_name"]
    else:
        devices[key][1]="Unknown"
    #print (key , '-->', value)

configfile.close()

# Print summary
for key,value in devices.items():
    if devices[key][0] == "Router" and not devices[key][2]:
        missingRouters[key] = devices[key][1]
    elif devices[key][0] == "EndDevice" and not devices[key][2]:
        missingEndDevices[key] = devices[key][1]
    if devices[key][0] == "Unknown":
        additionalDevices[key] = devices[key][1]

if len(missingRouters) > 0:
    print ("The following routers are not correctly set up, please remove them from Z2M config and execute the pairing operation again")
    for key, value in missingRouters.items():
        print ("\t" + key + " - " + value)
    if len(missingEndDevices) > 0:
        print ("Some end devices are missing too but you must first fix the routers")
elif len(missingEndDevices) > 0:
    print ("The following devices are not correctly set up, please remove them from Z2M config and execute the pairing operation again")
    for key, value in missingEndDevices.items():
        print ("\t" + key + " - " + value)
else :
   print ("Routers and end devices are correctly setup on the controller")

if len(additionalDevices) > 0:
   print ("Some devices are knwown by the controller but not present on your Zigbee network")
   for key, value in additionalDevices.items():
       print ("\t" + key)
````

PS: you will have to install python with the json and yaml modules.
I did this script for Z2M running on Linux, if you are using Windows, you will have to update the paths
PPS: make sure you have a recent coordinator_backup.json file, if you have the last Z2M version, this file is created every few hours, if you are running an older version, please restart Z2M before starting the script to trigger a backup (or upgrade to the latest Z2M version)

**Comment by mihaimdinca** [comment]
Thank you very much. I'll give it a try in the following days.

**Comment by RuinedOne** [comment]
@Fabiancrg Thanks for your script. I was going nuts trying to figure out why nothing would route/pair w/anything but the coordinator (Sonoff CC2652) and this solved my issue.

**Comment by Fabiancrg** [comment]
Hi
I put this for information but I don't now how to clean these devices.
I am not sure it is causing any issue so I did not investigate further.

**Comment by rhvs** [comment]
> 

I tried to edit your script to work in a HA OS... still getting some errors - would be great if you could have a look!

`import json
import yaml


devices = {}     # Dictionnary with all devices
missingRouters = {}
missingEndDevices = {}
additionalDevices = {}

backupfile = open('/config/zigbee2mqtt/coordinator_backup.json',"r")

DBfile = open('/config/zigbee2mqtt/database.db', 'r')
Lines = DBfile.readlines()


# Parse database.db to retrive all devices IEEE address and type define in Z2M

print('Parsing database.db file')

for line in Lines:
    DBdata = json.loads(line)
    if "ieeeAddr" in DBdata and "type" in DBdata:
        if DBdata ["type"] != 'Coordinator':
            devices[DBdata["ieeeAddr"]] = [DBdata ["type"], "", False]
            #print(DBdata["ieeeAddr"] + " is a " + DBdata ["type"])

DBfile.close()
print('Parsing database.db file... Done')

# Parse coordinator_backup.json to retreive all IEEE present in coordinator

data = json.loads(backupfile.read())


for device in data['devices']:
     dev = devices.get("0x" + device['ieee_address'])
     if dev:
         devices["0x" + device['ieee_address']][2] = True
     else:
         devices["0x" + device['ieee_address']]=["Unknown","", True]

backupfile.close()
print('Parsing backup file... Done')

# Retreive friendly name from configuration.yaml

configfile = open('/config/zigbee2mqtt/configuration.yaml',"r")

configData = yaml.safe_load(configfile)
configDevices = configData["devices"]


for key,value in devices.items():
    dev = configDevices.get(key)
    if dev:
        devices[key][1]=configDevices[key]["friendly_name"]
    else:
        devices[key][1]="Unknown"
    #print (key , '-->', value)

configfile.close()

# Print summary
for key,value in devices.items():
    if devices[key][0] == "Router" and not devices[key][2]:
        missingRouters[key] = devices[key][1]
    elif devices[key][0] == "EndDevice" and not devices[key][2]:
        missingEndDevices[key] = devices[key][1]
    if devices[key][0] == "Unknown":
        additionalDevices[key] = devices[key][1]

if len(missingRouters) > 0:
    print ("The following routers are not correctly set up, please remove them from Z2M config and execute the pairing op>
    for key, value in missingRouters.items():
        print ("\t" + key + " - " + value)
    if len(missingEndDevices) > 0:
        print ("Some end devices are missing too but you must first fix the routers")
elif len(missingEndDevices) > 0:
    print ("The following devices are not correctly set up, please remove them from Z2M config and execute the pairing op>
    for key, value in missingEndDevices.items():
        print ("\t" + key + " - " + value)
else :
   print ("Routers and end devices are correctly setup on the controller")

if len(additionalDevices) > 0:
   print ("Some devices are knwown by the controller but not present on your Zigbee network")
   for key, value in additionalDevices.items():
       print ("\t" + key)`

**Comment by rhvs** [comment]
@Fabiancrg 

I tried to edit your code to use it directly in HA OS.... but no luck (I have to admit it was a long shot :) ). Would be great if you could have a look!

```
import json
import yaml


devices = {}     # Dictionnary with all devices
missingRouters = {}
missingEndDevices = {}
additionalDevices = {}

backupfile = open('/config/zigbee2mqtt/coordinator_backup.json',"r")

DBfile = open('/config/zigbee2mqtt/database.db', 'r')
Lines = DBfile.readlines()


# Parse database.db to retrive all devices IEEE address and type define in Z2M

print('Parsing database.db file')

for line in Lines:
    DBdata = json.loads(line)
    if "ieeeAddr" in DBdata and "type" in DBdata:
        if DBdata ["type"] != 'Coordinator':
            devices[DBdata["ieeeAddr"]] = [DBdata ["type"], "", False]
            #print(DBdata["ieeeAddr"] + " is a " + DBdata ["type"])

DBfile.close()
print('Parsing database.db file... Done')

# Parse coordinator_backup.json to retreive all IEEE present in coordinator

data = json.loads(backupfile.read())


for device in data['devices']:
     dev = devices.get("0x" + device['ieee_address'])
     if dev:
         devices["0x" + device['ieee_address']][2] = True
     else:
         devices["0x" + device['ieee_address']]=["Unknown","", True]

backupfile.close()
print('Parsing backup file... Done')

# Retreive friendly name from configuration.yaml

configfile = open('/config/zigbee2mqtt/configuration.yaml',"r")

configData = yaml.safe_load(configfile)
configDevices = configData["devices"]

for key,value in devices.items():
    dev = configDevices.get(key)
    if dev:
        devices[key][1]=configDevices[key]["friendly_name"]
    else:
        devices[key][1]="Unknown"
    #print (key , '-->', value)

configfile.close()

# Print summary
for key,value in devices.items():
    if devices[key][0] == "Router" and not devices[key][2]:
        missingRouters[key] = devices[key][1]
    elif devices[key][0] == "EndDevice" and not devices[key][2]:
        missingEndDevices[key] = devices[key][1]
    if devices[key][0] == "Unknown":
        additionalDevices[key] = devices[key][1]

if len(missingRouters) > 0:
    print ("The following routers are not correctly set up, please remove them from Z2M config and execute the pairing op>
    for key, value in missingRouters.items():
        print ("\t" + key + " - " + value)
    if len(missingEndDevices) > 0:
        print ("Some end devices are missing too but you must first fix the routers")
elif len(missingEndDevices) > 0:
    print ("The following devices are not correctly set up, please remove them from Z2M config and execute the pairing op>
    for key, value in missingEndDevices.items():
        print ("\t" + key + " - " + value)
else :
   print ("Routers and end devices are correctly setup on the controller")

if len(additionalDevices) > 0:
   print ("Some devices are knwown by the controller but not present on your Zigbee network")
   for key, value in additionalDevices.items():
       print ("\t" + key)
```

**Comment by MnM001** [comment]
Hi,

@Fabiancrg  I have tried this script with my Sonoff ZB Dongle-E. 

It gave me a list of routers that needed to be "fixed". That list looked suspiciously like all the routers I have in my environment.
So I took one router out (held the button until it left my z2m environment). Requested a new backup and then I run the script again. That router was gone from the list of routers that needed to be "fixed".

I waited a few minutes and then added the router back.

After I requested a new backup I run the script again and the router that I removed and re-added again is in the list of  routers that needs to be "fixed".

How can that be? I just removed it (by button so the network key was deleted) and then add it back. It should had been fixed I assume.

Is the script not working as it should with the Sonoff ZB Dongle-E? 
Or I am doing something wrong?

**Comment by Koenkk** [comment]
The check is now supported directly from Zigbee2MQTT: https://github.com/Koenkk/zigbee2mqtt/pull/18599
